metagraph/src/cli/align.cpp

```diff
@@ -91,11 +91,8 @@ void map_sequences_in_file(const std::string &file,
     // TODO: multithreaded
     std::ignore = std::tie(thread_pool, print_mutex);
 
-    const DBGSuccinct *dbg = dynamic_cast<const DBGSuccinct*>(&graph);
-    if (!dbg) {
-        if (const auto *canonical = dynamic_cast<const CanonicalDBG*>(&graph))
-            dbg = dynamic_cast<const DBGSuccinct*>(&canonical->get_graph());
-    }
+    // used for sub-k sequence matching
+    const DBGSuccinct *dbg = dynamic_cast<const DBGSuccinct*>(&graph.get_base_graph());
```

Comment:
pass DBGSuccinct to the function. This conversion should happen in the caller. And in many cases it will be avoided there.

Otherwise you potentially enforce unnecessary conversions `DBGSuccinct -> DeBruijnGraph -> DBGSuccinct`